### ISSUE TITLE
Duplicate trace events are being fired

### ISSUE THREAD

**Issue by kenzieschmoll** [flutter/flutter#47020]
Example:

I have seen this with Duration events as well as Async events. Sometimes we see an extra begin event, an extra end event, or both.

```
{"name":"PipelineItem","cat":"Dart","tid":250009,"pid":249458,"ts":267504008453,"ph":"e","id":"1124","args":{"isolateId":"isolates/1022783825870871"}},
{"name":"PipelineItem","cat":"Dart","tid":250009,"pid":249458,"ts":267504008453,"ph":"e","id":"1124","args":{"isolateId":"isolates/1022783825870871"}},
```

**Comment by kenzieschmoll** [comment]
This is happening pretty frequently. Another example from today. Sometimes there can be other events sandwiched in between the duplicates.

```
{"name":"GPURasterizer::Draw","cat":"Embedder","tid":22275,"pid":42223,"ts":41314299193,"ph":"B","args":{}},{"name":"PipelineConsume","cat":"Embedder","tid":22275,"pid":42223,"ts":41314299193,"ph":"B","args":{}},{"name":"GPURasterizer::Draw","cat":"Embedder","tid":22275,"pid":42223,"ts":41314299193,"ph":"B","args":{}},
...
{"name":"GPURasterizer::Draw","cat":"Embedder","tid":22275,"pid":42223,"ts":41314306818,"ph":"E","args":{}},{"name":"PipelineItem","cat":"Embedder","tid":22275,"pid":42223,"ts":41314306818,"ph":"e","id":"20c","args":{}},{"name":"GPURasterizer::Draw","cat":"Embedder","tid":22275,"pid":42223,"ts":41314306818,"ph":"E","args":{}}
...

**Comment by kenzieschmoll** [comment]
@chinmaygarde

**Comment by TahaTesser** [comment]
Hi @kenzieschmoll 
Are you still experiencing this issue? 

If yes, Can you please provide your `flutter doctor -v` with the latest `stable` and `flutter run --verbose` logs, and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)
Thank you

**Comment by TahaTesser** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

**Comment by kenzieschmoll** [comment]
I am still experiencing this issue. This is causing issues with rendering the timeline in DevTools. We are having to use fragile logic to attempt to detect duplicates and remove them. Can someone from the engine team look into this and investigate how multiple events could be getting sent from the engine?

**Comment by kenzieschmoll** [comment]
Attached are some trace examples. `bad-trace-simple` is a set of raster events from GPU::RasterizerDraw begin to GPU::RasterizerDraw end for a single flutter frame. `bad-trace-very-simple` is a simplified version of `bad-trace-simple` that is easier to follow and see duplicates.

[bad-trace-simple.json.txt](https://github.com/flutter/flutter/files/6710009/bad-trace-simple.json.txt)
[bad-trace-very-simple.json.txt](https://github.com/flutter/flutter/files/6710008/bad-trace-very-simple.json.txt)

@chinmaygarde @iskakaushik

**Comment by chinmaygarde** [comment]
I don't think we are using the timeline API incorrectly. But, I also don't think the timeline provides guarantees about whether these events are sanitized in any way. AFAIK, all it cares about is that Catapult renders these traces correctly.

**Comment by kenzieschmoll** [comment]
If the Engine is sending multiple timeline events, wouldn't that be a bug? I'd be skeptical that the Timeline and TimelineTask APIs are sending duplicates.
